### PR TITLE
[ML-5816] Pregel python API

### PR DESCRIPTION
**Pull request by WeichenXu123**
Add python API for pregel.

### CODE REVIEW

File: python/graphframes/pregel.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        :param colName: the column name in the edge columns.
+        """
+        return col("edge." + colName)
+
+
+
+
+
+
+
```
Comment (felixcheung) [inline python/graphframes/pregel.py]: empty lines?

File: python/graphframes/__init__.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
 
-from .graphframe import GraphFrame
+from .graphframe import GraphFrame, Pregel
 
-__all__ = ['GraphFrame']
+__all__ = ['GraphFrame', 'Pregel']
```
Comment (mengxr) [inline python/graphframes/__init__.py]: Do not import Pregel by default (consistent with other algorithms).

File: python/graphframes/pregel.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    def __init__(self, gf):
+        super(Pregel, self).__init__()
+        self._java_obj = self._new_java_obj("org.graphframes.Pregel", gf._jvm_graph)
+
+    def setMaxIter(self, value):
+        """
+        Set max iteration number for the pregel running. Default value is 10.
+        """
+        self._java_obj.setMaxIter(value)
```
Comment (mengxr) [inline python/graphframes/pregel.py]: better use `int(..)`

File: python/graphframes/tests.py [hunk 4]
```diff
@@ -212,6 +212,36 @@ def test_bfs(self):
...
+                                       [2L, 0L],
+                                       [3L, 4L], # 3 has no in-links
+                                       [4L, 0L],
+                                       [4L, 2L]], ["src", "dst"])
+        edges.cache()
+        vertices = spark.createDataFrame([[0L], [1L], [2L], [3L], [4L]], ["id"])
+        numVertices = vertices.count()
+        vertices = GraphFrame(vertices, edges).outDegrees
+        graph = GraphFrame(vertices, edges)
+        pr_alpha = 0.85
```
Comment (mengxr) [inline python/graphframes/tests.py]: "alpha = 0.15"

**Comment by mengxr** [comment]
I made a small change to __init__.py. Now waiting for Travis.

**Pull request merged**